### PR TITLE
feat: upgrade to redwood

### PR DESCRIPTION
**Pull request by Henrrypg**
# Description

This PR does several changes to make it tutor friendly and upgrade to work with MySQL 8 and Mongo 7:

- changes to Dockerfile to install MySQL 8, Mongo7 clients
- Set the init jobs properly as tutor does
- add cronjobs to `k8s-jobs` patch

### CODE REVIEW

**Comment by MoisesGSalas** [comment]
Can we do the semantic + pyproject in a different PR?

File: drydock_backups/patches/k8s-jobs [hunk 1]
```diff
@@ -152,3 +152,18 @@ spec:
...
+  labels:
+    app.kubernetes.io/component: job
+spec:
+  ttlSecondsAfterFinished: 100
+  template:
+    spec:
+      restartPolicy: Never
+      containers:
+      - name: mongodb
+        image: {{ DOCKER_IMAGE_MONGODB }}
```
Comment (MoisesGSalas) [inline drydock_backups/patches/k8s-jobs:169]: What are we using this job for?

Comment (Henrrypg) [reply drydock_backups/patches/k8s-jobs:169]: Is the template to load [this job](https://github.com/eduNEXT/drydock-backups/pull/6/files#diff-c29b52c098c0d4017c4220701b2fead863755038c94927810dadfd58b1d5dba5R103). drydock is adding one mongodb job template but is to remove the dependency with drydock

**Comment by jfavellar90** [comment]
@Henrrypg could you please rebase?

**Comment by Henrrypg** [comment]
done @jfavellar90

File: drydock_backups/templates/drydock_backups/build/backups/Dockerfile [hunk 2]
```diff
@@ -1,17 +1,35 @@
-FROM mysql:5.7-debian as base
+FROM debian:bookworm-slim as base
```
Comment (MoisesGSalas) [inline drydock_backups/templates/drydock_backups/build/backups/Dockerfile]:
```suggestion
FROM debian:bookworm-slim AS base
```

File: drydock_backups/templates/drydock_backups/build/backups/Dockerfile [hunk 3]
```diff
@@ -1,17 +1,35 @@
...
+	gpgconf --kill all; \
+	rm -rf "$GNUPGHOME"
+
+RUN echo 'deb [ signed-by=/etc/apt/keyrings/mysql.gpg ] http://repo.mysql.com/apt/debian/ bookworm mysql-8.4-lts' > /etc/apt/sources.list.d/mysql.list
 
 # The following command is required to prevent the following error:
 # GPG error: http://repo.mysql.com/apt/debian buster InRelease: The following signatures couldn't be verified because the public key is not available: NO_PUBKEY B7B3B788A8D3785C
 # The solution was found here: https://askubuntu.com/questions/1497140/how-to-fix-apt-update-after-to-upgrade-the-mysql-list-file#comment2623210_1497141
-RUN gpg --keyserver hkp://keyserver.ubuntu.com:80 --recv-keys B7B3B788A8D3785C && rm /etc/apt/keyrings/mysql.gpg && gpg  --output /etc/apt/keyrings/mysql.gpg --export B7B3B788A8D3785C
+# RUN gpg --keyserver hkp://keyserver.ubuntu.com:80 --recv-keys B7B3B788A8D3785C && rm /etc/apt/keyrings/mysql.gpg && gpg  --output /etc/apt/keyrings/mysql.gpg --export B7B3B788A8D3785C
```
Comment (MoisesGSalas) [inline drydock_backups/templates/drydock_backups/build/backups/Dockerfile:23]: If this is no longer needed we can delete the whole comment block right?

File: drydock_backups/templates/drydock_backups/build/backups/Dockerfile [hunk 4]
```diff
@@ -1,17 +1,35 @@
...
 
-RUN apt-get update && apt-get install -y wget gnupg2
+RUN apt-get update && apt-get install -y wget gnupg2 curl
```
Comment (MoisesGSalas) [inline drydock_backups/templates/drydock_backups/build/backups/Dockerfile]: Why don't add this to the first apt-get at the start of the file? i assume we can drop gnupg if we are using gnupg2 right?

**Review by MoisesGSalas — APPROVED** [review]

File: drydock_backups/plugin.py [hunk 5]
```diff
@@ -99,3 +99,10 @@
     encoding="utf-8",
 ) as fi:
 
+
```
Comment (jfavellar90) [inline drydock_backups/plugin.py]: The 2 empty lines [here](https://github.com/eduNEXT/drydock-backups/blob/hpg/redwood-upgrade/drydock_backups/plugin.py#L101-L102) are strange, could we remove them?

**Review by jfavellar90 — APPROVED** [review]

**Comment by jfavellar90** [comment]
@Henrrypg I just left a comment. Anyways I give this PR thumbs up once the comment is addressed

**Pull request merged**